pymeasure/instruments/oxfordinstruments/itc503.py

```diff
@@ -196,30 +229,47 @@ def wait_for_temperature(self, error=0.01, timeout=3600,
                                     system to thermalize.
         :param should_stop: Optional function (returning a bool) to allow the
                             waiting to be stopped before its end.
+        :param max_comm_errors: The maximum number of communication errors that
+                                are allowed before the wait is stopped. if set
+                                to zero, no maximum will be used.
         """
 
         number_of_intervals = int(stability_interval / check_interval)
         stable_intervals = 0
         attempt = 0
+        comm_errors = 0
 
         t0 = time()
         while True:
-
-            if abs(self.temperature_error) < error:
-                stable_intervals += 1
+            try:
+                temp_error = self.temperature_error
+            except ValueError:
+                comm_errors += 1
+                log.error(
+                    "No temperature-error returned. "
+                    "Communication error # %d." % comm_errors
+                )
             else:
-                stable_intervals = 0
-                attempt += 1
+                if abs(temp_error) < error:
+                    stable_intervals += 1
+                else:
+                    stable_intervals = 0
+                    attempt += 1
 
             if stable_intervals >= number_of_intervals:
                 break
 
-            if timeout >= 0 and (time() - t0) > timeout:
+            if timeout > 0 and (time() - t0) > timeout:
                 raise TimeoutError(
                     "Timeout expired while waiting for the Oxford ITC305 to \
                     reach the set-point temperature"
                 )
 
+            if max_comm_errors > 0 and comm_errors > max_comm_errors:
```

Comment:
Now max_comm_errors == 0 means that an infinite amount errors is allowed, but now it is not possible to disallow any errors

Reply:
Maybe a better approach: use as kwarg default `max_comm_errors=None`, check with `if max_comm_errors is not None ...`. then it becomes possible to have max_comm_errors=0.